### PR TITLE
Add possibility to add review to a company.

### PR DESCRIPTION
**Pull request by laurandarius**

### CODE REVIEW

**Comment by laurandarius** [comment]
[Add the possibility to add a review to a company.](https://trello.com/c/3WL6gXwi/10-add-the-possibility-to-add-a-review-to-a-company)

File: app/components/CompanyReview/index.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Rating from 'material-ui-rating';
+
+import Line from 'components/Line/Loadable';
+import FormComponent from 'components/FormComponent';
+
+const CompanyReview = props => {
+  const {
+    hasReviewed,
```
Comment (lgnstn) [inline app/components/CompanyReview/index.js]: Please rename it `wasReviewed`.

File: app/containers/CompanyReview/actions.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  type: ADD_REVIEW_SUCCESS,
+  review,
+});
+
+export const addReviewFailure = errorMessage => ({
+  type: ADD_REVIEW_FAILURE,
+  errorMessage,
+});
+
+export const setHasReviewed = () => ({
```
Comment (lgnstn) [inline app/containers/CompanyReview/actions.js]: I don't think you need this action. You can set this flag when add review success happens.

File: app/containers/CompanyReview/index.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    comment: '',
+  },
+  validationSchema: Yup.object().shape({
+    comment: Yup.string('').required('This field is required.'),
+  }),
+  fields: [
+    {
+      type: 'input',
+      options: {
+        type: 'comment',
```
Comment (lgnstn) [inline app/containers/CompanyReview/index.js]: Shouldn't type be `text`?

File: app/containers/CompanyReview/index.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    canComment: false,
+    rating: 0,
+  };
+
+  onRating(rating) {
+    const { state } = this;
+    this.setState({
+      canComment: true,
+      rating,
+      ...state,
```
Comment (lgnstn) [inline app/containers/CompanyReview/index.js:60]: Have you tested this? If you add this line here, the last one, it should overwrite the previous lines, so the state will never change.

File: app/containers/CompanyReview/index.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    });
+  }
+
+  render() {
+    return (
+      <CompanyReviewComponent
+        onSubmit={values =>
+          this.props.onSubmit({
```
Comment (lgnstn) [inline app/containers/CompanyReview/index.js]: When you use props values, define them as the first thing in the function, like this: `const { onSubmit, loggedUserId, companyId, rating, ... } = this.props;`.

File: app/containers/CompanyReview/reducer.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+});
+
+const addReviewReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_REVIEW_SUCCESS:
+      return state.set('errorMessage', '');
+    case ADD_REVIEW_FAILURE:
+      return state.set('errorMessage', action.errorMessage);
+    case SET_HAS_REVIEWED:
+      return state.set('hasReviewed', action.hasReviewed);
```
Comment (lgnstn) [inline app/containers/CompanyReview/reducer.js]: This value should be saved in the company. Maybe you should update the API so it returns whether the logged use has reviewed a company or not.

File: app/containers/CompanyReview/reducer.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+import {
+  ADD_REVIEW_SUCCESS,
+  ADD_REVIEW_FAILURE,
+  SET_HAS_REVIEWED,
+} from './constants';
+
+const initialState = fromJS({
+  errorMessage: '',
+  hasReviewed: false,
+  canComment: false,
```
Comment (lgnstn) [inline app/containers/CompanyReview/reducer.js]: This shouldn't be in redux. We talked about this and agreed to keep this information in the component state.

File: app/containers/CompanyReview/saga.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+import addReviewRequest from 'services/addCompanyReviewRequest';
+import { ADD_REVIEW } from './constants';
+import { addReviewSuccess, addReviewFailure, setHasReviewed } from './actions';
+
+export function* addReview(action) {
+  const { data } = action;
+  const response = yield addReviewRequest(data);
+  if (response.id) {
+    yield put(addReviewSuccess(response));
+    yield put(setHasReviewed());
```
Comment (lgnstn) [inline app/containers/CompanyReview/saga.js]: As I said before, you don't need this one.

**Review by lgnstn — CHANGES_REQUESTED** [review]

**Comment by laurandarius** [comment]
@lgnstn updated.

**Review by lgnstn — APPROVED** [review]

**Pull request merged**